core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -979,6 +979,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     // A Hadoop configuration can be about 10 KB, which is pretty big, so broadcast it.
     val confBroadcast = broadcast(new SerializableConfiguration(hadoopConfiguration))
     val setInputPathsFunc = (jobConf: JobConf) => FileInputFormat.setInputPaths(jobConf, path)
+    clean(setInputPathsFunc)
```

Comment:
OK yeah it's added back for `hadoopFile` calls but not others, like `hadoopRDD` or direct use. I don't know enough to evaluate this change with authority. If it's there for correctness it has to be there, and this does look like a function that needs cleaning, right? It may not happen to need cleaning in the case you're exercising but that doesn't mean it works to not clean it.